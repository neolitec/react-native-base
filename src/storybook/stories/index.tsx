import React from 'react';
import { storiesOf } from '@storybook/react-native/dist';
import { View, Text, ViewStyle } from 'react-native';

const style = {
  flex: 1,
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#F5FCFF',
};

const CenteredView = ({ children }: { children: React.ReactNode }) => (
  <View style={style as ViewStyle}>{children}</View>
);

storiesOf('CenteredView', module).add('default view', () => (
  <CenteredView>
    <Text>Hello Storybook</Text>
  </CenteredView>
));
