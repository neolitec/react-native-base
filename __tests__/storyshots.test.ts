import initStoryshots from '@storybook/addon-storyshots';

initStoryshots({
  framework: 'react-native',
  config: ({ configure }: any) =>
    configure(() => {
      // eslint-disable-next-line global-require
      require('../src/storybook/stories');
    }, module),
});
