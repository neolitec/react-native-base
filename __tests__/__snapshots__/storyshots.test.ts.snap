// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots CenteredView default view 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello Storybook
  </Text>
</View>
`;
